React/Views/RCTView.m

```diff
@@ -118,6 +118,11 @@ - (instancetype)initWithFrame:(CGRect)frame
     _hitTestEdgeInsets = UIEdgeInsetsZero;
 
     _backgroundColor = super.backgroundColor;
+
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(voiceOverStatusDidChange:)
+                                                 name:UIAccessibilityVoiceOverStatusChanged
+                                               object:nil];
```

Comment:
RN produces quite large view hierarchies. thousands might be a little aggressive, but hundreds definitely happens quite often.